src/clinical/clinical-service.ts

```diff
@@ -324,14 +334,69 @@ export const getDonorSubmissionErrorUpdates = async (
     submitterDonorIds: [],
   };
 
-  await donorDao.findByPaginatedProgramId(programId, errorQuery).then(donorData => {
-    validDonors = donorData.donors
-      .filter(donor => donor.schemaMetadata.isValid)
-      .map(({ donorId }) => Number(donorId));
-  });
+  const donorData = (await donorDao.findByPaginatedProgramId(programId, errorQuery)).donors;
+
+  const validDonorIds = donorData
+    .filter(donor => donor.schemaMetadata.isValid)
+    .map(({ donorId }) => Number(donorId));
+
+  const clinicalErrors: ClinicalErrorsResponseRecord[] = [];
+
+  const invalidDonorIds = errorDonorIds
+    .filter(donorId => !validDonorIds.includes(donorId))
+    .filter((entity, index, array) => array.indexOf(entity) === index);
+
+  for (const donorId of invalidDonorIds) {
+    const currentDonor = donorData.filter(donor => donor.donorId === donorId)[0];
+    const { submitterId: submitterDonorId } = currentDonor;
+    const currentDonorErrors = clinicalMigrationErrors
+      .filter(errorRecord => errorRecord.donorId === donorId)
+      .map(migrationError => migrationError.errors)
+      .flat();
+
+    const currentDonorEntities = currentDonorErrors
+      .map(error => error.entityName)
+      .filter((entity, index, array) => array.indexOf(entity) === index);
```

Comment:
same comment as above - this filter function will give back original array

Reply:
same filtering function which could be created once as util function `filterDuplicates` and used like `filter(filterDuplicates)`

Reply:
Yes duplicates were a problem so this was necessary, could definitely extract into reusable function

Reply:
Updated w/ reusable filter